From 522d68e1495766aeb7cf629d581d9dfeb21bd3b4 Mon Sep 17 00:00:00 2001
From: Peter Geis <pgwipeout@gmail.com>
Date: Thu, 25 Mar 2021 20:56:04 -0400
Subject: [PATCH] arm64: dts: rockchip: fixup quartz64 gpio regulators

Change from regulator-gpio to regulator-fixed.
This solves issues with random failure to enumerate regulators.

Signed-off-by: Peter Geis <pgwipeout@gmail.com>
---
 .../boot/dts/rockchip/rk3566-quartz64a.dts     | 24 ++++++++-----------
 1 file changed, 10 insertions(+), 14 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-quartz64a.dts b/arch/arm64/boot/dts/rockchip/rk3566-quartz64a.dts
index 47c8f0e9bcb0..9a7e5d57db27 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-quartz64a.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-quartz64a.dts
@@ -49,26 +49,22 @@
 			<0x0 0xfd90f000 0x0 0x1000>;
 	};
 
-	pcie20_3v3: gpio-regulator {
-		compatible = "regulator-gpio";
+	pcie20_3v3: pcie20_3v3 {
+		compatible = "regulator-fixed";
 		regulator-name = "pcie20_3v3";
-		regulator-min-microvolt = <0100000>;
+		regulator-min-microvolt = <3300000>;
 		regulator-max-microvolt = <3300000>;
-		gpios = <&gpio0 RK_PC2 GPIO_ACTIVE_HIGH>;
-		gpios-states = <0x1>;
-		states = <0100000 0x0
-			  3300000 0x1>;
+		gpio = <&gpio0 RK_PC2 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
 	};
 
-	vcc5v0_host: gpio-regulator {
-		compatible = "regulator-gpio";
+	vcc5v0_host: vcc5v0_host {
+		compatible = "regulator-fixed";
 		regulator-name = "vcc5v0_host";
-		regulator-min-microvolt = <0100000>;
+		regulator-min-microvolt = <5000000>;
 		regulator-max-microvolt = <5000000>;
-		gpios = <&gpio4 RK_PB5 GPIO_ACTIVE_HIGH>;
-		gpios-states = <0x1>;
-		states = <0100000 0x0
-			  5000000 0x1>;
+		gpio = <&gpio4 RK_PB5 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
 	};
 
 	ebc_dev: ebc-dev {
-- 
GitLab

